### DISCUSSION TITLE
Signing Messages with signMessage/signMessageWith6492 SDK function on Ethereum Chain results in error

### DISCUSSION THREAD

**Discussion by HatimPatwa** [alchemyplatform/aa-sdk#967] [Issues]
### Issue Description

When calling **signMessage** or **signMessageWith6492** sdk methods exposed by creating modular account alchemy client
throws back error specifically on ***Ethereum Network***, rest networks work fine and return signed message. 

Eg. raw message passed : 0x4d7920656d61696c206973206a6f686e40646f652e636f6d202d2031373234393332363931323430

### Error 
> Redacted API keys for sharing here

```

URL: https://eth-mainnet.g.alchemy.com/v2/<Alchemy Key>
Request body: {"method":"eth_call","params":[{"data":"0x84b0196e","from":"0xB20Bfc9680D87F18ECd52eD76B3ddCb3b8D08d0b"},"latest"]}
 
Raw Call Arguments:
  from:  0xB20Bfc9680D87F18ECd52eD76B3ddCb3b8D08d0b
  data:  0x84b0196e
 
Contract Call:
  function:  eip712Domain()

Docs: https://viem.sh/docs/contract/readContract
Details: Missing or invalid parameters.
Double check you have provided the correct parameters.

URL: https://eth-mainnet.g.alchemy.com/v2/<Alchemy Key>
Request body: {"method":"eth_call","params":[{"data":"0x84b0196e","from":"0xB20Bfc9680D87F18ECd52eD76B3ddCb3b8D08d0b"},"latest"]}

Details: Missing or invalid parameters.
Double check you have provided the correct parameters.

URL: https://eth-mainnet.g.alchemy.com/v2/<Alchemy Key>
Request body: {"method":"eth_call","params":[{"data":"0x84b0196e","from":"0xB20Bfc9680D87F18ECd52eD76B3ddCb3b8D08d0b"},"latest"]}

Details: stack underflow (0 <=> 5)
Version: viem@2.9.9
Version: viem@2.9.9
Version: viem@2.9.9
```

### Code snippet

```js

//dynamic selecting network based on request
const network = EIP155_MAINNET_CHAINS[body.chainId];

//Our modular client
 const smartAccountClient = await createModularAccountAlchemyClient({
    apiKey: network.api_key,
    chain: network.chain,
    signer: signer,
  });

//just logging for track
  logger.log('info', `-------smart address ${smartAccountClient.account.address}------------`)

  let signedMessage: Hex; //type safety
  try {
//function to sign 
    signedMessage = await smartAccountClient.signMessageWith6492({
      account: smartAccountClient.account,
      message: {
        raw: transactionData.message, //the message in Hex
      }
    });

//example message passed above
//0x4d7920656d61696c206973206a6f686e40646f652e636f6d202d2031373234393332363931323430

//just for verification on our end
    const verifyMsg2 = await smartAccountClient.verifyMessage({
      address: smartAccountClient.account.address,
      message: {
        raw: transactionData.message,
      },
      signature: signedMessage,
    });

    logger.log('info', `-------signed message 2------------${verifyMsg2}`)

  } catch (error) {
    console.log('error', error)
    logger.log('error', `-------sign message failed------------ \n ${(error)}`)
    throw new ApiError(400, 'Transaction failed, please try again later')
  }

  return signedMessage
  ```
  
  ### SDK versions
  
 ```
"@alchemy/aa-accounts": "^3.6.1",
"@alchemy/aa-alchemy": "^3.7.3",
"@alchemy/aa-core": "^3.6.1",
"alchemy-sdk": "^3.2.0",
    
```

**Comment by moldy530** [comment]
ah, I think there might be 1 of 2 (or 2 of 2) things happening here. Can you try to verify the message with a client that's NOT the `smartAccountClient`? you can use `createPublicClient` for example to verify messages. Due to a bug in `viem` verifying messages with the `smartAccountClient` that generated the message, will result in failure.

The second issue, might be based on how we generate the signature, but I want to eliminate the first possibility

**Reply by HatimPatwa** [reply]
Hi @moldy530 , thanks for the response.

To  eliminate the first possibility, i have to generate a signature first right?

Thats what i am not able to do, The error occurs during the function call of signMessage
at below code snippet:
```js
//function to sign 
    signedMessage = await smartAccountClient.signMessageWith6492({
      account: smartAccountClient.account,
      message: {
        raw: transactionData.message, //the message in Hex
      }
    });
```

Hope this helps to narrow down the issue!

**Reply by moldy530** [reply]
if the client already has an account on it, then you shouldn't need to have this line: `account: smartAccountClient.account`

Also if possible, can you upgrade to the latest versions of the SDK? we're on v4 now, but looking at your v4 versions we had a couple bug fixes post the one's you're on. Think we got as far as `3.18.1` in v3, but I can't remember exactly

the v4 migration guide is here: https://accountkit.alchemy.com/migration-guide

**Reply by HatimPatwa** [reply]
Ok great will do. though it will take time because current architecture users v3 all over place 😄

**Reply by moldy530** [reply]
you can also try to update to the latest v3 versions, we did have some fixes around signing messages with accounts before migrating to v4

**Reply by HatimPatwa** [reply]
No Worries , we are already onramping to V4 , since it has so many improvement to account abstraction and we don't wanna miss it.
Thanks a lot @moldy530
